### PR TITLE
fix: top margin for switch, checkbox & radio button

### LINKED ISSUE #402: Radio button: Text a few pixels too high up
## Description

When compared to the designs the Label for radio buttons on mobile seems to be two pixels above where the designed label is. On desktop the text is 1 pixel above where it should be.

## Steps to reproduce

1. Go to http://opensource.corp.adobe.com/spectrum-css/
2. Click on ...
3. Observe ...

## Screenshots

###Mobile screenshot
image

###Desktop screenshot
image

## Environment
 - **Spectrum CSS version:** 
 - **Browser(s) and OS(s):** 
Found on https://reactspectrum.blob.core.windows.net/reactspectrum/20e5131cfc8f65620868f16dc32defaedd215aaf/index.html?path=/story/radio--default

### LINKED ISSUE #403: Switch label spacing slightly off
## Description

This is the same problem as on the [Radio button component label bug](https://github.com/adobe/spectrum-css/issues/402). On the the Label for switches on mobile it seems to be two pixels above where the designed label is. On desktop the text is 1 pixel above where it should be.

## Steps to reproduce

1. Go to http://opensource.corp.adobe.com/spectrum-css/
2. Click on ...
3. Observe ...

## Screenshots

Desktop 
image

Mobile
image

## Environment
 - **Spectrum CSS version:** 
 - **Browser(s) and OS(s):** 

https://reactspectrum.blob.core.windows.net/reactspectrum/20e5131cfc8f65620868f16dc32defaedd215aaf/index.html?path=/story/switch--default

### LINKED ISSUE #406: Checkbox label position 1px/2px too low
## Description
Checkbox label is 1px (desktop) / 2px (mobile) lower than the design

## Steps to reproduce

1. Go to https://opensource.adobe.com/spectrum-css/components/checkbox/
2. Screenshot checkbox
3. Compare with UI kit

## Expected behavior
Label should be 1px higher (desktop) or 2px higher (mobile)

## Screenshots
(red is design)
desktop:
checkbox-desktop

mobile:
checkbox-mobile

## Environment
 - **Spectrum CSS version:** 
 - **Browser(s) and OS(s):** 

## Additional context
Came across while doing React review. Not sure if this is some rendering issue or not :( sorry if this is not relevant!

### PR DESCRIPTION
**Pull request by bernhard-adobe**
Correcting the top margin for checkboxes & radio buttons. 
https://jira.corp.adobe.com/browse/SDS-4420 closes #406, 
https://jira.corp.adobe.com/browse/SDS-4418 closes #402,
https://jira.corp.adobe.com/browse/SDS-4403 closes #403 

## Description
Adding the correct margin. There seems to be a "wrapping study" in #176 that hard-coded the value `--spectrum-global-dimension-size-75`. The correct value should be `--spectrum-global-dimension-size-85`.
@lazd  Maybe we should remove the hard-coded value instead if the wrapping study is complete.  

## How and where has this been tested?
Go to the following pages and inspect the label for the checkbox or radio button
http://localhost:3000/docs/radio.html
http://localhost:3000/docs/checkbox.html
http://localhost:3000/docs/toggle.html

Browser(s) and OS(s): macOS Mojave 10.14.5 (18F203)
Google Chrome Version 80.0.3968.0 (Official Build) canary (64-bit)

## Screenshots
**The orange background is for reference only to show the span of the margin applied to the switch, textbox and radio button label**

Before:

Screen Shot 2019-11-21 at 16 18 39
Screen Shot 2019-11-21 at 16 14 21
Screen Shot 2019-11-22 at 11 05 29

After:
Screen Shot 2019-11-21 at 16 18 50
Screen Shot 2019-11-21 at 16 14 29
Screen Shot 2019-11-22 at 11 05 42

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).

- [x] This pull request is ready to merge.

### CODE REVIEW

**Comment by bernhard-adobe** [comment]
sounds good to me @lazd . Feel free reference the Jira ticket number here.

**Comment by lazd** [comment]
@bernhard-adobe go ahead and ref this issue in the comments and we can merge this! https://jira.corp.adobe.com/browse/SDS-4466

**Comment by lazd** [comment]
@bernhard-adobe please test your fix in Edge and Firefox per this comment: https://github.com/adobe/spectrum-css/issues/406#issuecomment-562343426

**Comment by lazd** [comment]
@bernhard-adobe sounds like we're not ready to merge given the Firefox and Edge issues. Let me know if you need some backup on this issue and I can jump in to help.

**Comment by bernhard-adobe** [comment]
@lazd  I proposed a fix for the inconsistent line-height fix in #406. 
Feel free to let us know how you'd like to proceed.

**Comment by lazd** [comment]
@bernhard-adobe that fix is acceptable to me. Waiting on @GarthDB and @jianliao to weigh in.

File: components/radio/index.css [hunk 1]
```diff
@@ -79,9 +81,12 @@ governing permissions and limitations under the License.
 .spectrum-Radio-label {
   margin-left: var(--spectrum-radio-text-gap);
   font-size: var(--spectrum-radio-text-size);
-
   transition: color var(--spectrum-global-animation-duration-100) ease-in-out;
 
+  /* Fix for inconsistent line-height between browsers that would push the label 1px below the intended baseline */
+  /* more info: https://stackoverflow.com/questions/47700568/css-fonts-render-differently-in-firefox-and-chrome */
+  line-height: 1.49;
```
Comment (bernhard-adobe) [inline components/radio/index.css]: Move to top of the root.

**Comment by bernhard-adobe** [comment]
Also tested this on Firefox vs Chrome in large mode and it looks good.
Screenshot: 

Screen Shot 2019-12-11 at 13 56 38

**Comment by lazd** [comment]
@bernhard-adobe excellent, this is good to go. Thanks!

**Pull request merged**